client/mysql.cc

```diff
@@ -1408,8 +1408,11 @@ sig_handler handle_sigint(int sig)
   /* terminate if no query being executed, or we already tried interrupting */
   if (!executing_query || (interrupted_query == 2))
   {
-    tee_fprintf(stdout, "Ctrl-C -- exit!\n");
-    goto err;
+    tee_fprintf(stdout, "Ctrl-C\n");
+    rl_on_new_line(); // Regenerate the prompt on a newline
+    rl_replace_line("", 0); // Clear the previous text
+    rl_redisplay();
+    return;
   }
```

Comment:
the condition and comment say "if no query being executed OR we already tried interrupting".

I think that in the second case it's pretty weird to do what you're doing, there's no prompt to regenerate.

Probably, the correct behavior would be still to quit the client. Or, may be better, to disconnect and show the prompt again.

Reply:
Updated according to your inputs, thanks!